src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -41,24 +41,32 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
         {
             // Verify the credentials are correct.
             // Have to actually ping a storage operation.
-            IStorageBlobClient client = account.CreateBlobClient();
+            IStorageQueueClient client = account.CreateQueueClient();
 
             try
             {
                 // This can hang for a long time if the account name is wrong. 
                 // If will fail fast if the password is incorrect.
-                await client.GetServicePropertiesAsync(cancellationToken);
+                await client.SdkObject.GetServicePropertiesAsync(cancellationToken);
             }
             catch (OperationCanceledException)
             {
                 throw;
             }
-            catch
+            catch (StorageException exception)
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                                        "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                if (exception.IsRemoteNameNotResolved())
```

Comment:
Validator is used for all storage accounts yep.  @lindydonna had mentioned that we don't want to support blob-only, but I've lifted that restriction for secondary storage accounts with this new iteration (since they work at the moment, no need for a breaking change).